src/GSF.CarbonAware/src/Handlers/IEmissionsHandler.cs

```diff
@@ -1,7 +1,27 @@
+using GSF.CarbonAware.Models;
+
 namespace GSF.CarbonAware.Handlers
 {
     public interface IEmissionsHandler
     {
+        /// <summary>
+        /// Calculate the observed emission data by list of locations for a specified time period.
+        /// </summary>
+        /// <param name="locations">String array of named locations, must be non-empty (ex: ["eastus"])</param>
+        /// <param name="start">[Optional] Start time for the data query. (ex: 2022-03-01T15:30:00Z)</param>
+        /// <param name="end">[Optional] End time for the data query. (ex: 2022-03-01T18:30:00Z)</param>
+        /// <returns>Array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(string[] locations, DateTimeOffset? start = null, DateTimeOffset? end = null);
+
+        /// <summary>
+        /// Calculate the best emission data by list of locations for a specified time period.
+        /// </summary>
+        /// <param name="locations">String array of named locations, must be non-empty (ex: ["eastus"])</param>
+        /// <param name="start">[Optional] Start time for the data query. (ex: 2022-03-01T15:30:00Z)</param>
+        /// <param name="end">[Optional] End time for the data query. (ex: 2022-03-01T18:30:00Z)</param>
+        /// <returns>An array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetBestEmissionsDataAsync(string[] locations, DateTimeOffset? start = null, DateTimeOffset? end = null);
```

Comment:
I don't think it makes sense for this to just take a single location. The goal of this function is to find the location that has the best emissions data so having only one is basically just the equivalent of calling `GetEmissionsDataAsync`

Reply:
Having only one location will return the lowest value in the time range, where `GetEmissionsDataAsync` will return all data points.

The most likely use-case would be to use this to calculate potential savings if workloads were moved to the best time in a past range.  Ideally one would use the `ForecastByDate` function for that, but this would be a reasonable next choice if the provider does not have that (EG - Json data or ElectricityMaps). Given that we take a single location there, it feels reasonable to take one here as well.

Reply:
Oh I see, I didn't realize that slightly different behavior, that makes sense why we want that overload now. Thanks